src/components/ProcessMoneyRequestHoldMenu.tsx

```diff
@@ -31,6 +32,14 @@ function ProcessMoneyRequestHoldMenu({isVisible, onClose, onConfirm, anchorPosit
     const {translate} = useLocalize();
     const styles = useThemeStyles();
     const popoverRef = useRef(null);
+    const navigation = useNavigation();
+
+    useEffect(() => {
+        const unsub = navigation.addListener('beforeRemove', () => {
+            onClose();
+        });
+        return unsub;
+    }, [navigation, onClose]);
```

Comment:
Can you leave a comment why we need this useEffect?
Also I thought that modal will be unmounted when its parent's unmounted, do we really need this one?